actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -653,10 +653,32 @@ def convert_value_to_parameters(value)
         end
       end
 
+      def permit_nested_array(elements, object)
+        permitted = []
+        
+        object.each do |nested_object|
+          case nested_object
+          when Array
+            permitted << permit_nested_array(elements, nested_object)
+          else
+            permitted << nested_object.select {|nested| elements.keys.include?(nested)}
+          end
+        end
+        
+        permitted
+      end
+      
       def each_element(object)
         case object
         when Array
-          object.grep(Parameters).map { |el| yield el }.compact
+          allowed_elements = object.flatten.grep(Parameters).map { |el| yield el }.compact
+          
+          if object.first.is_a?(Array)
+            permit_nested_array(allowed_elements.first, object)
```

Comment:
I'm not sure if I understand your point,
but if we always calls `permit_nested_array` even object is not a nested array, it will need to permit again the parameter, for example: 

if we change the code
```
allowed_elements = object.flatten.grep(Parameters).map { |el| yield el }.compact
 if object.first.is_a?(Array)
  permit_nested_array(allowed_elements.first, object)
 else
    allowed_elements
  end
```

for something like

```
allowed_elements = object.flatten.grep(Parameters).map { |el| yield el }.compact
allowed_elements.contact(allowed_elements, object)
```
when it's not a nested array, the method will try to permit again the elements,

please let me know your opinion, or if it was a missunderstood it,

Thanks